controllers/front/product.php

```diff
@@ -45,16 +45,27 @@ public function initContent()
         if ($token === false) {
             exit();
         }
+        ProcessLoggerHandler::openLogger();
+        $etag = sprintf('W/"%s"', ShoppingfeedPreloading::getEtag($token['id_shoppingfeed_token']));
+        header('Etag: ' . $etag);
+        if ($_SERVER['HTTP_IF_NONE_MATCH'] === $etag) {
+            header('HTTP/1.1 304 not modified');
```

Comment:
Please use `http_response_code` instead, the server will manage the HTTP version.